active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,320 @@
+
+### Enabling input validation
+
+Setting `ValidationMode` to `InvalidInputOverwritten` will enable NumberBox to overwrite invalid input that is not numerical nor legally formulaic with the last valid value when evaluation is triggered on loss of focus or a press of the "Enter" key.
+
+XAML
+```XAML
+<NumberBox Header="Quantity"
+    Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}"
+    ValidationMode="InvalidInputOverwritten" />
+```
+
+Setting `ValidationMode` to `Disabled` allows custom input validation to be configured.  
+
+### Formatting input 
+
+[Number formatting](https://docs.microsoft.com/en-us/uwp/api/windows.globalization.numberformatting) can be used to format the value of a Numberbox by configuring an instance of a formatting class and assigning it to the `NumberFormatter` property. Decimal, currency, percent, and significant figures are few of the number formatting classes available. 
+
+Here is an example of using DecimalFormatter to format a NumberBox's value to have one integer digit and two fraction digits:  
+
+XAML
+```XAML
+<NumberBox  x:Name="FormattedNumberBox"
+Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}" />
+```
+
+C#
+```C#
+private void SetNumberBoxNumberFormatter()
+{
+    DecimalFormatter formatter = new DecimalFormatter();
+    formatter.IntegerDigits = 1;
+    formatter.FractionDigits = 2;
+    FormattedNumberBox.NumberFormatter = formatter;
+}
+```
+
+![A NumberBox showign a value of 0.00.](images/numberbox-formatted.PNG)
+
+### Enabling hyper scroll 
+
+Setting `IsHyperScrollEnabled` to true enables the value of NumberBox to be incremented or decremented by scrolling when a NumberBox has focus and is being hovered over. 
+
+XAML
+```XAML
+<NumberBox Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}"
+    IsHyperScrollEnabled="True" />
+```
+
+## Remarks
+
+### Input Scope
+
+`Number` will be used for the [input scope](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Input.InputScopeNameValue). This input scope is intended for working with digits 0-9. This may be overwritten but alternative InputScope types will not be explicitly supported. 
+
+### Not a Number
+
+When a NumberBox is cleared of input, `Value` will be set to `NaN` to indicate no numerical value is present. 
+
+## Recommendations
+
+* `Text` and `Value` make it easy to capture the value of a NumberBox as a String or as a Double without needing to convert the value bewteen types. When programmatically altering the value of a NumberBox, it is recommended to do so through the `Value` property. `Value` will overwrite `Text` in initial set up. After the initial set up, changes to one will be progrogated to the other, but consistently making programmatic changes through `Value` helps avoid any conceptual misunderstanding that NumberBox will accept non-numeric characters through `Text`.  
```

Comment:
What happens if I set Text to something that's clearly not a number?  What parser are you using to convert `box.Text = "0.2"` into a numeric value?

Reply:
Discussion: This is an example of an invalid expression. Current behavior is that it will revert back to previous text.

Reply:
How do I the developer know that my "x.Text = StringFromSomeFileOrJsonBlob" didn't take?  How would I pre-validate the string before handing it to the Text property?

Reply:
The most straightforward option would be to monitor the `ValueChanged` event or `Value` property to ensure the intended value took. 

Alternatively, I would encourage the developer to leverage our [Input Validation](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/lucashaines/inputvalidation/active/InputValidation/InputValidation.md) work, once available, to ensure they have full control of the input validation configuration and events.